lib/smart_proxy_dns_powerdns/dns_powerdns_configuration_validator.rb

```diff
@@ -0,0 +1,37 @@
+require 'smart_proxy_dns_powerdns/dns_powerdns_plugin'
+
+module Proxy::Dns::Powerdns
+  class ConfigurationValidator
+    def validate_settings!
+      validate_choice(:powerdns_backend, ['mysql', 'postgresql', 'dummy'])
+
+      case settings[:powerdns_backend]
+      when 'mysql' then
+        validate_presence :powerdns_mysql_username, :powerdns_mysql_password, :powerdns_mysql_database
+      when 'postgresql' then
+        validate_presence :powerdns_postgresql_connection
```

Comment:
Out of curiosity, why do you have settings for username and password for mysql, but none for pg, which supports those as well (https://github.com/ged/ruby-pg/blob/master/lib/pg/connection.rb#L24)? If you decide to add these for pg, you could keep the same setting names for both dbs, which might make validation a bit easier...